src/Adapter/Currency/CommandHandler/ToggleExchangeRateAutomatizationHandler.php

```diff
@@ -0,0 +1,262 @@
+        TranslatorInterface $translator,
+        $isCronJobModuleInstalled,
+        $adminBaseUrl,
+        $dbPrefix
+    ) {
+        $this->configuration = $configuration;
+        $this->contextShop = $contextShop;
+        $this->isCronJobModuleInstalled = $isCronJobModuleInstalled;
+        $this->shopUrlDataProvider = $shopUrlDataProvider;
+        $this->adminBaseUrl = $adminBaseUrl;
+        $this->dbPrefix = $dbPrefix;
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws CurrencyException
+     */
+    public function handle(ToggleExchangeRateAutomatizationCommand $command)
+    {
+        if (!$this->isCronJobModuleInstalled) {
+            throw new ScheduleExchangeRatesUpdateException(
+                'Live exchange rates feature cannot be modified because "cronjob" module is not installed',
+                ScheduleExchangeRatesUpdateException::CRON_TASK_MANAGER_MODULE_NOT_INSTALLED
+            );
+        }
+
+        $this->configuration->restrictUpdatesTo($this->contextShop);
+
+        $cronId = (int) $this->configuration->get('PS_ACTIVE_CRONJOB_EXCHANGE_RATE');
+        $thereIsOneCronRunning = ($cronId !== 0);
+
+        try {
+            if ($thereIsOneCronRunning && $command->exchangeRateStatus()) {
+                $this->removeConfigurationIfNotFoundOrIsDeactivated($cronId);
+            }
+
+            if (!$thereIsOneCronRunning && $command->exchangeRateStatus()) {
+                $this->enableExchangeRatesScheduler();
+            }
+
+            if ($thereIsOneCronRunning && !$command->exchangeRateStatus()) {
+                $this->disableExchangeRatesScheduler($cronId);
+            }
+        } catch (Exception $exception) {
+            throw new CurrencyException(
+                'An unexpected error occurred when trying to update live exchange rates',
+                0,
+                $exception
+            );
+        }
+    }
+
+    /**
+     * @return string
+     *
+     * @throws PrestaShopException
+     */
+    private function getCronUrl()
+    {
+        $protocol = Tools::getShopProtocol();
+        $shopDomain = $this->shopUrlDataProvider->getMainShopUrl()->domain;
+        $cronFileLink = sprintf(
+            'cron_currency_rates.php?secure_key=%s',
+            md5($this->configuration->get('_COOKIE_KEY_') . $this->configuration->get('PS_SHOP_NAME'))
+        );
+
+        return $protocol . $shopDomain . $this->adminBaseUrl . $cronFileLink;
+    }
+
+    /**
+     * @param string $cronUrl
+     *
+     * @return bool
+     *
+     * @throws Exception
+     */
+    private function createCronJob($cronUrl)
+    {
+        /** @var CronJobs $cronJobsModule */
+        $cronJobsModule = Module::getInstanceByName('cronjobs');
+
+        $isCronAdded = $cronJobsModule->addOneShotTask(
+            $cronUrl,
+            $this->translator->trans(
+                'Live exchange Rate for %shop_name%',
```

Comment:
Is the capital letter here necessary? I know it was the way we used to write this wording but I would prefer _Live exchange rate for %shop_name%_ instead, in case it does not change anything. ;-)